src/main/java/org/elasticsearch/search/highlight/Highlighter.java

```diff
@@ -26,4 +28,6 @@
     String[] names();
 
     HighlightField highlight(HighlighterContext highlighterContext);
+
+    public boolean cannotHighlight(FieldMapper fieldMapper);
```

Comment:
Ha! Just saw that @eebbrr started working on that already a year ago and did the exact same thing! You are two already so yeah, I'll rename to `canHighlight`.